packages/gatsby-source-contentful/src/normalize.js

```diff
@@ -563,17 +519,39 @@ exports.createNodesForContentType = ({
           fieldType === `RichText` &&
           _.isPlainObject(entryItemFields[entryItemFieldKey])
         ) {
-          const richTextNode = prepareRichTextNode(
-            entryNode,
-            entryItemFieldKey,
-            entryItemFields[entryItemFieldKey],
-            createNodeId
-          )
+          const fieldValue = entryItemFields[entryItemFieldKey]
+
+          const rawReferences = []
+
+          // Locate all Contentful Links within the rich text data
+          const traverse = obj => {
+            for (let k in obj) {
+              const v = obj[k]
+              if (v && v.sys && v.sys.type === `Link`) {
+                rawReferences.push(v)
+              } else if (v && typeof v === `object`) {
+                traverse(v)
+              }
+            }
+          }
 
-          childrenNodes.push(richTextNode)
-          entryItemFields[`${entryItemFieldKey}___NODE`] = richTextNode.id
+          traverse(fieldValue)
```

Comment:
I hope these nodes aren't particularly big. This will be pretty expensive at scale if these objects are large and deep.

Reply:
It depends on the content. They should not have to many object/array values, but the values itself might go big. (aka the text in the paragraphs and so on)

Reply:
The scalar size (string len) is not the problem here. Okay :)